# try 语句和异常处理

## 简介

典型的异常包括失去数据库连接以及遇到意外输入等，异常处理机制为程序中异常检测和异常处理两部分的协作提供支持：

* throw 表达式：throw 表达式用于表示它遇到了无法处理的问题，我们说 throw 引发了异常
* try 语句块：异常处理部分使用 try 语句块处理异常，它以关键字 try 开始，并以一个或多个 catch 子句结束
* 异常类：用于在 throw 表达式和 catch 子句之间传递异常的具体信息

## throw 表达式

程序的异常检测部分用 throw 表达式抛出一个异常：

```c++
#include <stdexcept>

throw runtime_error("tomocat");
```

## try 语句块

try 语句块的语法是：

> Tips：try 语句块内声明的变量在块外部无法访问，特别是在 catch 子句内也无法访问。

```c++
try {
 program-statements
} catch (exception-declaration) {
 handler-statements
} catch (exception-declaration) {
 handler-statements
}
```